### ISSUE TITLE
kern messages logged not according to rsyslog.conf

### ISSUE THREAD

**Issue by testrsyslog** [rsyslog/rsyslog#5311]
### Expected behavior
When rsyslog.conf file contains the following lines :
*.err;\
kern.warn;\
user.notice	/var/log/messages

kern.warn messages should be logged in /var/log/messages.
kern.notice messages should **not** be logged  in /var/log/messages.

### Actual behavior

With the previously mentionned lines, both kern.warn and kern.notice messages are logged in /var/log/messages

Note: i do not use any additional rsyslog configuration file with $IncludeConfig

### Steps to reproduce the behavior

Replace all lines corresponding to "facility.priority file" in /etc/rsyslog.conf by : 
*.err;\
kern.warn;\
user.notice	/var/log/messages

Save the file and restart rsyslog service

Run the following commands:
logger -p kern.warn test_kern_warn
logger -p kern.notice test_kern_notice
"test_kern_warn" is correctly logged in /var/log/messages.
"test_kern_notice" is also logged in /var/log/messages but shouldn't.

### Environment
- rsyslog version: 8.2102.0-5.el8
- platform: AlmaLinux 8.5

**Comment by rgerhards** [comment]
use this instead (one line):
```
*.err;kern.warn;user.notice /var/log/messages
```

does that change anything?

**Comment by testrsyslog** [comment]
> use this instead (one line):
> 
> ```
> *.err;kern.warn;user.notice /var/log/messages
> ```
> 
> does that change anything?

No, the behavior is the same. I still produce kern.notice lines in /var/log/messages

**Comment by rgerhards** [comment]
Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html

**It's sufficient to see the startup only, you can shutdown rsyslog immediately after it has fully started up**. Once this is done, please post the debug log (e.g. via pastebin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

**Comment by testrsyslog** [comment]
> Please create a debug log. Instructions: https://www.rsyslog.com/doc/v8-stable/troubleshooting/howtodebug.html
> 
> **It's sufficient to see the startup only, you can shutdown rsyslog immediately after it has fully started up**. Once this is done, please post the debug log (e.g. via pastebin or github). After that, we can review the contents and check how exactly rsyslog processing worked.

Here's the debug log : 
[rsyslog.debug.txt](https://github.com/rsyslog/rsyslog/files/13953507/rsyslog.debug.txt)

I started the rsyslog service and did the 3 commands:
logger -p kern.err "test_kern_err"
logger -p kern.warn "test_kern_warn"
logger -p kern.notice "test_kern_notice"

**Comment by rgerhards** [comment]
I didn't notice it previously, but you cannot generate kernel messages with the logger tool. Looking at the debug log, I see that the facility is always 1 (user). Example:

```
<13>Jan 16 16:58:44 snet: test_kern_notice
```

From the CLI, you can use

```
echo "<message text>" > /dev/kmsg
```

If running as root. But I right now do not know how to set the priority in this case.

**Comment by rgerhards** [comment]
In other words: there is no bug in rsyslog, but in the testing ;-)

**Comment by rgerhards** [comment]
Yeah, the debug log tell this approach does not work:

```
ruleset.c: processBATCH: next msg 0:  testwarn
```

The PRI is stripped. Not sure if the kernel does this our systemd journal is in the way. IMHO the only real way to verify this is by writing a program that calls `printk()` - or wait for real kernel messages.

**Comment by rgerhards** [comment]
> I tried to send messages by affecting a priority this way :
> echo '<4> testwarn' >> /dev/kmsg
> echo '<5> testnotice' >> /dev/kmsg

yes - but it did not work. You may want to check with the journal or your kernel settings. These are still user-level message. Something (not rsyslog) changes the priority.
